### PR TITLE
[WIP/DNM]: feat(kobo): enable syncing of annotations to Hardcover

### PR DESCRIPTION
**Pull request by black-dragon74**
### Actively being tested and updated. DO NOT REVIEW.

**Linked Issue:** Fixes #

> **Required.** Every PR must reference an approved issue. If no issue exists, [open one](https://github.com/booklore-app/booklore/issues/new) and wait for maintainer approval before submitting a PR. Unsolicited PRs without a linked issue will be closed.

## 🏷️ Type of Change

- [ ] Bug fix
- [ ] New feature
- [ ] Enhancement to existing feature
- [ ] Refactor (no behavior change)
- [ ] Breaking change (existing functionality affected)
- [ ] Documentation update

## 🔧 Changes

-

## 🧪 Testing (MANDATORY)

> **PRs without this section filled out will be closed.** "Tests pass" or "Tested locally" is not sufficient. You must provide specifics.

**Manual testing steps you performed:**

1.
2.
3.

**Regression testing:**

-

**Edge cases covered:**

-

**Test output:**

<details>
<summary>Backend test output (<code>./gradlew test</code>)</summary>

```
PASTE OUTPUT HERE
```

</details>

<details>
<summary>Frontend test output (<code>ng test</code>)</summary>

```
PASTE OUTPUT HERE
```

</details>

## 📸 Screen Recording / Screenshots (MANDATORY)

> Every PR must include a **screen recording or screenshots** showing the change working end-to-end in a running local instance (both backend and frontend). This means you must have actually built, run, and tested the code yourself. PRs without visual proof will be closed without review.

---

## ✅ Pre-Submission Checklist

> **All boxes must be checked before requesting review.** Incomplete PRs will be closed without review. No exceptions.

- [ ] This PR is linked to an approved issue
- [ ] Code follows project style guidelines and conventions
- [ ] Branch is up to date with `develop` (merge conflicts resolved)
- [ ] I ran the full stack locally (backend + frontend + database) and verified the change works
- [ ] Automated tests added or updated to cover changes (backend **and** frontend)
- [ ] All tests pass locally and output is pasted above
- [ ] Screen recording or screenshots are attached above proving the change works
- [ ] PR is a single focused change (one bug fix OR one feature, not multiple unrelated changes)
- [ ] PR is reasonably scoped (PRs over 1000+ changed lines will be closed, split into smaller PRs)
- [ ] No unsolicited refactors, cleanups, or "improvements" are bundled in
- [ ] Flyway migration versioning is correct _(if schema was modified)_
- [ ] Documentation PR submitted to [booklore-docs](https://github.com/booklore-app/booklore-docs) _(if user-facing changes)_

### 🤖 AI-Assisted Contributions

> **If any part of this PR was generated or assisted by AI tools (Copilot, Claude, ChatGPT, etc.), all items below are mandatory.** You are fully responsible for every line you submit. "The AI wrote it" is not an excuse, and AI-generated PRs that clearly haven't been reviewed are the #1 reason PRs get closed.

- [ ] I have read and understand every line of this PR and can explain any part of it during review
- [ ] I personally ran the code and verified it works (not just trusted the AI's output)
- [ ] PR is scoped to a single logical change, not a dump of everything the AI suggested
- [ ] Tests validate actual behavior, not just coverage (AI-generated tests often assert nothing meaningful)
- [ ] No dead code, placeholder comments, `TODO`s, or unused scaffolding left behind by AI
- [ ] I did not submit refactors, style changes, or "improvements" the AI suggested beyond the scope of the issue

---

### CODE REVIEW

File: booklore-api/src/main/java/org/booklore/controller/KoboReadingServicesController.java [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+ * Transparent proxy for Kobo reading services requests.
+ * <p>
+ * Kobo devices send reading services requests (annotations, user storage, etc.)
+ * directly to the host, ignoring any path in {@code reading_services_host}.
+ * These arrive as {@code /api/v3/...} and {@code /api/UserStorage/...} and must
+ * be proxied to {@code readingservices.kobo.com}.
+ */
+@Slf4j
+@RestController
+public class KoboReadingServicesController {
```
Comment (black-dragon74) [inline booklore-api/src/main/java/org/booklore/controller/KoboReadingServicesController.java:29]: @acx10 @balazs-szucs Need a little help here. In my testing, I am able to sync the annotations to hardcover.

Still, the device always fails with "sync failed, try again later".

Pretty sure I am messing something up when proxy-ing the request to Kobo. Any pointers?

_This feature is almost complete (in its initial form), just this once thing needs ironing out._

If it helps, I have a KLC, any way I can see why the sync fails?

Comment (black-dragon74) [reply booklore-api/src/main/java/org/booklore/controller/KoboReadingServicesController.java:29]: @acx10 ^^

**Comment by nicr9** [comment]
Hey, is this something that should be broken up into two different features?

Kobo -> booklore annotations
booklore annotations -> hardcover (maybe with options to opt-in by book/library)

I've got a Kobo and I've got a Hardcover account. I can see the value in making it easy to push out my thoughts on a book but I'm not sure I'd like to automatically publish all of my annotations.
